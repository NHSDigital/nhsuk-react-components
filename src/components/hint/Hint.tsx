import React, { HTMLProps, useContext } from 'react';

import FormGroupContext from '../formgroup/FormGroupContext';
import classNames from 'classnames';

export type HintProps = HTMLProps<HTMLSpanElement>;

const Hint: React.FC<HintProps> = props => {
  const { id, className, ...rest } = props;
  const { inputID } = useContext(FormGroupContext);
  const inputId = id || inputID ? `${inputID}--hint` : undefined;
  return <span id={inputId} className={classNames('nhsuk-hint', className)} {...rest} />;
};

export default Hint;
