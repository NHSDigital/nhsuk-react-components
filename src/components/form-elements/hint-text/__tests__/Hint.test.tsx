import React from 'react';
import { render } from '@testing-library/react';
import Hint from '../';

describe('Hint', () => {
  it('matches snapshot', () => {
    const { container } = render(<Hint>Text</Hint>);

    expect(container).toMatchSnapshot('Hint');
  });

  it('renders children', () => {
    const { container } = render(<Hint>Text</Hint>);

    expect(container.textContent).toBe('Text');
  });
});
