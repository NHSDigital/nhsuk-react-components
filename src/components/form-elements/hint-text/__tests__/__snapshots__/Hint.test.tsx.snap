// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hint matches snapshot: Hint 1`] = `
<div>
  <div
    class="nhsuk-hint"
  >
    Text
  </div>
</div>
`;
